api/node/server/server.go

```diff
@@ -307,3 +314,46 @@ func (s *Server) GetHareBeaconEpoch(ctx context.Context,
 	}
 	return &beaconResp{b: beacon}, nil
 }
+
+type proposalResp struct {
+	buf   []byte
+	nonce types.VRFPostIndex
+}
+
+func (p *proposalResp) VisitGetProposalLayerNodeResponse(w http.ResponseWriter) error {
+	if p.buf == nil {
+		w.WriteHeader(204)
+		return nil
+	}
+	w.Header().Add("content-type", "application/octet-stream")
+	w.Header().Add("x-spacemesh-atx-nonce", fmt.Sprintf("%d", p.nonce))
+	w.WriteHeader(200)
+	_, err := w.Write(p.buf)
+	return err
+}
+
+func (s *Server) GetProposalLayerNode(ctx context.Context, request GetProposalLayerNodeRequestObject) (
+	GetProposalLayerNodeResponseObject, error,
+) {
+	hexBuf, err := hex.DecodeString(request.Node)
+	if err != nil {
+		return &proposalResp{}, err
+	}
+	id := types.BytesToNodeID(hexBuf)
+
+	proposal, nonce, err := s.proposals.BuildFor(ctx, types.LayerID(request.Layer), id)
+	if err != nil {
+		return &proposalResp{}, err
+	}
+	if proposal == nil && err == nil {
```

Comment:
The `err` is nil here always.
```suggestion
	if proposal == nil {
```